fk/secretsend_test.go

```diff
@@ -79,3 +88,266 @@ func assertMessageBodyMatchesSecretContent(unverifiedBody io.Reader, secret stri
+		"SEF":    "\033[ Y",  // Sheet Eject and Feed
+		"SGR":    "\033[m",   // Select Graphic Rendition
+		"SHS":    "\033[ K",  // Select Character Spacing
+		"SIMD":   "\033[^",   // Select Implicit Movement Direction
+		"SL":     "\033[ @",  // Scroll Left
+		"SLH":    "\033[ U",  // Set Line Home
+		"SLL":    "\033[ V",  // Set Line Limit
+		"SLS":    "\033[ h",  // Set Line Spacing
+		"SM":     "\033[h",   // Set Mode
+		"SOS":    "\033X",    // Start of String
+		"SPA":    "\033V",    // Start of Guarded Area
+		"SPD":    "\033[ S",  // Select Presentation Directions
+		"SPH":    "\033[ i",  // Set Page Home
+		"SPI":    "\033[ G",  // Spacing Increment
+		"SPL":    "\033[ j",  // Set Page Limit
+		"SPQR":   "\033[ X",  // Select Print Quality and Rapidity
+		"SR":     "\033[ A",  // Scroll Right
+		"SRCS":   "\033[ f",  // Set Reduced Character Separation
+		"SRS":    "\033[[",   // Start Reversed String
+		"SSA":    "\033F",    // Start of Selected Area
+		"SSU":    "\033[ I",  // Select Size Unit
+		"SSW":    "\033[ [",  // Set Space Width
+		"SS2":    "\033N",    // Single-Shift Two
+		"SS3":    "\033O",    // Single-Shift Three
+		"ST":     "\033\\",   // String Terminator
+		"STAB":   "\033[ ^",  // Selective Tabulation
+		"STS":    "\033S",    // Set Transmit State
+		"SU":     "\033[S",   // Scroll Up
+		"SVS":    "\033[ L",  // Select Line Spacing
+		"TAC":    "\033[ b",  // Tabulation Aligned Centred
+		"TALE":   "\033[ a",  // Tabulation Aligned Leading Edge
+		"TATE":   "\033[ `",  // Tabulation Aligned Trailing Edge
+		"TBC":    "\033[g",   // Tabulation Clear
+		"TCC":    "\033[ c",  // Tabulation Centred on Character
+		"TSR":    "\033[ d",  // Tabulation Stop Remove
+		"TSS":    "\033[ E",  // Thin Space Specification
+		"VPA":    "\033[d",   // Line Position Absolute
+		"VPB":    "\033[k",   // Line Position Backward
+		"VPR":    "\033[e",   // Line Position Forward
+		"VTS":    "\033J",    // Line Tabulation Set
+		"C0":     "\033!@",   // Control Set 0 Announcer
+		"C1":     "\033&@",   // Control Set 1 Announcer
+		"C1ALT1": "\033 F",   // Control Set 1 Announcer Alternate 1
+		"C1ALT2": "\033\"F",  // Control Set 1 Announcer Alternate 2
+	}
+
+	for name, code := range invalidANSICodes {
+		t.Run(fmt.Sprintf("for '%v')", name), func(t *testing.T) {
+			if containsDisallowedRune(code) != true {
+				t.Fatalf("expected '%v' (%v) to be a disallowed but it wasn't", code, name)
+			}
+		})
+	}
+
+	validANSICodes := map[string]string{
+		"LF": "\012", // Line Feed
+		"CR": "\015", // Carriage Return
+	}
+
+	for name, code := range validANSICodes {
+		t.Run(fmt.Sprintf("for '%v')", name), func(t *testing.T) {
+			if containsDisallowedRune(code) != false {
+				t.Fatalf("expected '%v' (%v) to be allowed but it wasn't", code, name)
+			}
+		})
+	}
+
+	for _, naughtyString := range naughtystrings.Unencoded() {
+		t.Run(fmt.Sprintf("'%v' doesn't crash it", naughtyString), func(t *testing.T) {
```

Comment:
I know I suggested using a new test runner for each string but I regret it since it spams the rest output in a way that's not particularly helpful!